### ISSUE TITLE
Unable to publish without index.html gives no feedback

### ISSUE THREAD

**Issue by gideonthomas** [mozilla/thimble.mozilla.org#2355]
Looks like a regression on both staging and prod!

STR:
1. Login and start a new project
2. Delete the index.html file
3. Click Publish

The publish dialog box vibrates but shows no explanation (which I think we had at the bottom of the box).

**Comment by gideonthomas** [comment]
cc @flukeout

**Comment by flukeout** [comment]
Hey @gideonthomas - do you mean click "Update Published" after deleting the index.html file?

**Comment by gideonthomas** [comment]
yeah sorry, I meant the "Update Published Version"

**Comment by humphd** [comment]
@flukeout @gideonthomas do you think this is a good bug for @MPierre9 to work on?  If so, can you provide some guidance on where to begin?

**Comment by flukeout** [comment]
Yeah, I think this would be a good one!

**Comment by MPierre9** [comment]
@humphd @flukeout Thanks! I'll get started :)

**Comment by MPierre9** [comment]
@flukeout I got Thimble running on my local machine using the instructions on GitHub, however when I go to make new project and recreate the bug it hangs on "Creating new project" am I missing a step or installation instruction. Thanks

Also is there a slack channel I can ask questions?

**Comment by humphd** [comment]
@MPierre9 hop into https://chat.mozillafoundation.org/mozilla/channels/thimble

**Comment by flukeout** [comment]
That's interesting - when you say it hangs, what are you seeing? Can you post any errors you are seeing in your console? Users run into this once in a while, so it's handy that you are seeing it too so maybe we can debug together.

**Comment by MPierre9** [comment]
It was stuck at loading project but I figured it out, I wasn't running the brackets server and thimble server at the same time *facepalm*

**Comment by humphd** [comment]
@MPierre9 yeah, that's something a lot of people run into.  One thing we could do is have the Thimble node.js server check that the Brackets host is available on startup, and print some kind of useful error message if it isn't.  If you want to file and fix that bug, I'll help you figure out what to do.

**Comment by MPierre9** [comment]
@humphd That sounds like a good idea I don't have a lot of experience with node.js but I am very familiar with other server side languages like PHP and am willing to jump in to fix that bug. Any pointers are appreciated :)

**Comment by humphd** [comment]
@MPierre9 I filed https://github.com/mozilla/thimble.mozilla.org/issues/2509 for you.

**Comment by humphd** [comment]
This is a test.

https://github.com/mozilla/thimble.mozilla.org/blob/master/server/index.js#L6-L9

**Comment by MPierre9** [comment]
Finally got some time to work on this bug, is this what you're looking for @gideonthomas ?

```
if (!bramble.hasIndexFile()) {
    // This width hack lets us apply the class again in a way that
    // re-triggers the animation.
    console.info("Inside has index condition");
    if (publisher.dialogEl.hasClass("cannot-publish")) {
      console.info("Inside has index condition TWO");
      
      publisher.dialogEl.removeClass("cannot-publish");
      publisher.dialogEl.width(publisher.dialogEl.width());
    }
    publisher.dialogEl.addClass("cannot-publish");
    $('#publish-buttons').show();
    return;
  }
```

![thimble index html fix](https://user-images.githubusercontent.com/27080760/31312704-6fb2902a-ab98-11e7-88c5-5c9f3eeba36c.gif)
